### PR TITLE
Build: Migrate from JSHint & JSCS to ESLint

### PR DESCRIPTION
**Pull request by mgol**
The work is divided into three commits with (mostly, there are exceptions) the following work:
1. The actual switch to ESLint + fixes done by ESLint autofix.
2. Fixing the `strict` rule.
3. Manually fixing remaining violations.
4. Fixing a test issue caused by introducing strict mode.

Previously, while JSCS was only running in the build process in the command line, JSHint was also running in TestSwarm. There's no official ESLint version running in the browser so I opted to drop this feature; Core doesn't use it and I don't think it's important enough to hold off the switch to ESLint; you will get nice feedback when building in the terminal anyway. Please let me know if that's not acceptable for some reason.

### CODE REVIEW

**Comment by fnagel** [comment]
>Previously, while JSCS was only running in the build process in the command line, JSHint was also running in TestSwarm. There's no official ESLint version running in the browser so I opted to drop this feature; Core doesn't use it and I don't think it's important enough to hold off the switch to ESLint

As long as the new ESLint runs with each build (and here in the GitHub / CI setup) I'm fine with the change.

File: tests/unit/tooltip/helper.js [hunk 1]
```diff
@@ -14,10 +14,8 @@ define( [
 						timers = jQuery.timers;
 
 					jQuery.fx.stop();
-					var x = false;
```
Comment (fnagel) [inline tests/unit/tooltip/helper.js:16]: This looks like a breaking change. Is variable `x` not in use?

**Review by fnagel — APPROVED** [review]
+1 by reading

Looks good expect for one thing (the removed x variable). Amazing how many formatting issues have been fixed, even though there have been CGL tests in place...

File: tests/unit/tooltip/helper.js (hunk 1 above)
Comment (mgol) [reply tests/unit/tooltip/helper.js:16]: It’s not used, that’s why ESLint complained. It’s only written to but never read.

Comment (fnagel) [reply tests/unit/tooltip/helper.js:16]: Ahh ok. Strange enough nobody noticed before...

**Pull request merged**